File: UsageMonitoring.App/Models/CodexQuotaSnapshot.cs
using System.Text.Json.Serialization;

namespace UsageMonitoring.App.Models;

public sealed record CodexQuotaSnapshot(
    IReadOnlyList<RateLimitBucket> Limits,
    CreditStatus? Credits,
    string? PlanType,
    DateTimeOffset SyncedAtUtc)
{
    public static CodexQuotaSnapshot Empty { get; } = new(
        Array.Empty<RateLimitBucket>(),
        Credits: null,
        PlanType: null,
        SyncedAtUtc: DateTimeOffset.MinValue);

    [JsonIgnore]
    public bool HasDisplayableData =>
        Limits.Count > 0 ||
        Credits is { Unlimited: true } ||
        Credits is { HasCredits: true };
}

File: UsageMonitoring.Mac/Sources/UsageMonitoringMac/Models/CodexQuotaSnapshot.swift
import Foundation

struct CreditStatus: Codable, Equatable, Sendable {
    let hasCredits: Bool
    let unlimited: Bool
    let balance: String?

    var shouldDisplay: Bool {
        unlimited || hasCredits
    }
}

struct CodexQuotaSnapshot: Codable, Equatable, Sendable {
    var limits: [RateLimitBucket]
    var credits: CreditStatus?
    var planType: String?
    var syncedAt: Date

    var hasDisplayableUsage: Bool {
        !limits.isEmpty || credits?.shouldDisplay == true
    }
}
